medusa/server/api/v2/internal.py

```diff
@@ -208,10 +208,11 @@ def resource_search_indexers_for_show_name(self):
         # Get all possible show ids
         all_show_ids = {}
         for show in app.showList:
-            for external in show.externals:
-                indexer_id = reverse_mappings.get(external)
-                show_id = show.externals[external]
-                all_show_ids.setdefault(indexer_id, []).append(show_id)
+            for ex_indexer_name, ex_show_id in iteritems(show.externals):
+                ex_indexer_id = reverse_mappings.get(ex_indexer_name)
+                if not ex_indexer_id:
+                    continue
+                all_show_ids[(ex_indexer_id, ex_show_id)] = (show.indexer_name, show.series_id)
```

Comment:
The changes made here are:
 - Iterating over the show externals more efficiently (using `iteritems` instead of `external` + `show.externals[external]`)
 - The dictionary structure is now:
```py
all_show_ids = {
    # Key: Show external value: (indexer id, show id)
    # Value: Show in library: (indexer name, show id) -- notice: name
    # For a show that was added from TVDB:
    (1, 123456) : ('tvdb', 123456),
    (3, 456123) : ('tvdb', 123456),
    (4, 516243) : ('tvdb', 123456),
    # ... for each show in the library
}
```